src/CoreWCF.Primitives/src/CoreWCF/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -298,7 +298,7 @@ protected override async Task<(Stream, SecurityMessageProperty)> OnAcceptUpgrade
                 try
                 {
                     SecurityToken token = new X509SecurityToken(certificate2, false);
-                    ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = _parent.ClientCertificateAuthenticator.ValidateToken(token);
+                    ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = _parent.ClientCertificateAuthenticator.ValidateTokenAsync(token).GetAwaiter().GetResult();
                     _clientSecurity = new SecurityMessageProperty
```

Comment:
This is actually a good example of why to keep the sync methods around. X409SecurityTokenAuthenticator doesn't actually do anything async in it's validation implementation, so we could avoid creating a Task, then getting the Awaiter and then getting the result and could directly call it here.